core/node/rpc/node2node_ephemeral.go

```diff
@@ -104,7 +125,40 @@ func (s *Service) saveEphemeralMiniblock(ctx context.Context, req *SaveEphemeral
 	}
 
 	// Save the ephemeral miniblock.
-	// Here we are sure that the record of the stream exists in the storage.
+	location, err := s.storage.GetMediaStreamLocation(ctx, streamId)
+	if err != nil {
+		return err
+	}
+	if location != s.externalMediaStorage.GetBucket() {
+		return fmt.Errorf("external media stream storage changed after this ephemeral media was created.")
+	}
+	if location != "" {
+		uploadID, partNum, err := s.storage.GetExternalMediaStreamNextPart(ctx, streamId)
+		if err != nil {
+			return err
+		}
+		etag, err := s.externalMediaStorage.UploadPartToExternalMediaStream(
+			ctx,
+			streamId,
+			storageMb.Data,
+			uploadID,
+			partNum,
+		)
+		if err != nil {
+			return err
+		}
+		if s.storage.WriteExternalMediaStreamPartInfo(
+			ctx,
+			streamId,
+			storageMb.Number,
+			partNum,
+			etag,
+			len(storageMb.Data),
+		) != nil {
+			return err
+		}
```

Comment:
This is not going to work as expected since `err` was not assigned to the given call result.